File: backend/app/models/platform.py
from datetime import datetime

from sqlalchemy import Integer, String, Float, Text, DateTime, ForeignKey, func, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Platform(Base):
    """平台/组织 — 用户可创建或加入的平台/圈子"""
    __tablename__ = "platforms"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(128), nullable=False, comment="平台名称")
    platform_no: Mapped[str | None] = mapped_column(String(32), unique=True, nullable=True, comment="平台编号")
    creator_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False, comment="创建者(秘书长)")
    annual_fee: Mapped[float] = mapped_column(Float, default=0.0, comment="年费")
    description: Mapped[str] = mapped_column(Text, default="", comment="平台描述")
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), onupdate=func.now())


class PlatformMember(Base):
    """平台成员 — 用户与平台的多对多关联，带角色"""
    __tablename__ = "platform_members"
    __table_args__ = (UniqueConstraint("platform_id", "user_id", name="uq_platform_user"),)

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    platform_id: Mapped[int] = mapped_column(Integer, ForeignKey("platforms.id"), nullable=False)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False)
    role: Mapped[str] = mapped_column(
        String(32), default="member",
        comment="角色: secretary_general(秘书长) / secretariat(秘书处) / member(会员)"
    )
    joined_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

File: liankebao-weapp/src/api/platform.ts
/**
 * 资源平台 API — 询赋吸收模块
 *
 * 对接后端: /api/business-card/platforms/*
 * 响应格式: { code: number, message: string, data: any }
 */

import { api } from './client'

export interface Platform {
  id: number
  name: string
  platform_no?: string
  creator_id: number
  annual_fee: number
  description: string
  member_count: number
  resource_count: number
  created_at: string
  updated_at: string
}

export interface PlatformMember {
  id: number
  user_id: number
  name: string
  company: string
  title: string
  avatar: string
  role: 'secretary_general' | 'secretariat' | 'member'
  joined_at: string
}

export interface PlatformCreateParams {
  name: string
  description?: string
  annual_fee?: number
  platform_no?: string
}

export interface PlatformReport {
  platform_id: number
  platform_name: string
  total_members: number
  role_distribution: {
    secretary_general: number
    secretariat: number
    member: number
  }
  coverage: {
    company_count: number
    industry_count: number
  }
}

const platformApi = {
  /** 平台推荐列表（按成员数排序） */
  list: (keyword?: string, skip = 0, limit = 20) => {
    const params: Record<string, any> = { skip, limit }
    if (keyword !== undefined) params.keyword = keyword
    return api.get<Platform[]>('/api/business-card/platforms', { data: params })
  },

  /** 平台详情 */
  getById: (id: number) => {
    return api.get<Platform>(`/api/business-card/platforms/${id}`)
  },

  /** 创建平台（创建者自动成为秘书长） */
  create: (data: PlatformCreateParams) => {
    return api.post<Platform>('/api/business-card/platforms', data)
  },

  /** 更新平台信息 */
  update: (id: number, data: Partial<PlatformCreateParams>) => {
    return api.put<Platform>(`/api/business-card/platforms/${id}`, data)
  },

  /** 成员列表（按角色排序） */
  getMembers: (platformId: number) => {
    return api.get<PlatformMember[]>(`/api/business-card/platforms/${platformId}/members`)
  },

  /** 加入平台（角色=member） */
  join: (platformId: number) => {
    return api.post<any>(`/api/business-card/platforms/${platformId}/join`)
  },

  /** 商业报告（仅秘书长可见） */
  getReport: (platformId: number) => {
    return api.get<PlatformReport>(`/api/business-card/platforms/${platformId}/report`)
  },
}

export default platformApi
